SimpleCalculator/SimpleCalculator/ViewController.swift

```diff
@@ -9,12 +9,143 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    
+    var result = 0.0
+    var pointInUse = false
+    var currentfuncType = ArithmeticFunctions.none
+    var error = false
+    var isResult = true
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
     }
-
-
+    
+    @IBOutlet weak var textField: UILabel!
+    
+    @IBAction func buttonsOnAction(_ sender: Any) {
+        let button = sender as! UIButton
+        
+        if error && button.restorationIdentifier != "buttonAC"{
+            return
+        }
+        
+        switch button.restorationIdentifier {
+        case "buttonAC":
```

Comment:
using string literals is error prone approach, if you need strings as keys you can wrap it with enums